### DISCUSSION TITLE
How to export the generated QR

### DISCUSSION THREAD

**Discussion by svprdga** [lukef/qr.flutter#124] [Q&A] [answered]
In the README it says: _Export to image data to save to file or use in memory_

I want to export the image to memory in order to be able to share it, in the README it doesn't say how to do it.

How can I export the generated QR?

**Comment by MahmedRahman** [comment, marked as answer]
i think tis artical can help 
https://medium.com/codex/exporting-qr-codes-in-flutter-dd30220fcba4

**Reply by sarankumar-ns** [reply]
The author made this article available to Medium members only.

**Reply by symbolisticism** [reply]
I also need help with this, and I'd prefer not to have to subscribe to Medium to know how to do it.

**Reply by frieder-audriga** [reply]
Because the article is not freely available, and I was not able to find this documented elsewhere: You need to create a `QrPainter` object and call `.toImageData` on it.

The one liner I will be using to get `Uint8List` is as follows:
```dart
Uint8List? qrBytes = (await QrPainter(data: data, version: QrVersions.auto).toImageData(2048))?.buffer.asUint8List();
```

**Comment by Heidelbergg** [comment]
Is a premium article, is there an unlocked or better version of it?

**Comment by symbolisticism** [comment]
Thank you so much!
